gateway/api/use_cases/dependencies_versions.py

```diff
@@ -0,0 +1,22 @@
+"""Authentication use case to manage the authentication process in the api."""
+
+import logging
+from typing import Optional
+from django.contrib.auth.models import AbstractUser
+
+from api.utils import create_dynamic_dependencies_whitelist
+
+
+logger = logging.getLogger("gateway.use_cases.authentication")
+
+
+class AvailableDynamicDependenciesUseCase:  # pylint: disable=too-few-public-methods
```

Comment:
We should probably modify the configuration for `too-few-public-methods` if not we are going to need to add it in every use case class.